server/routes/auth-routes.js

```diff
@@ -0,0 +1,85 @@
+const router = require('express').Router();
+const passport = require('passport');
+let client_add = "http://localhost:3000";
+
+// function ensureAuthenticated
+// ensures the user is logged in before it grants access to the api.
+// input: req, the request; res, the response obj; next, the function to be called next
+function ensureAuthenticated(req, res, next)
+{
+    if (req.isAuthenticated()) {
+        return next();
+    } else {
+        res.status(403).json( {"failure": "you are not logged in"});
```

Comment:
use 401 if not logged in. 403 is if you are authenticated but forbidden from accessing the resource. 
rename failure to err for consistency